kie-soup-maven-utils/kie-soup-maven-integration/src/test/java/org/appformer/maven/integration/ArtifactResolverTest.java

```diff
@@ -0,0 +1,77 @@
+package org.appformer.maven.integration;
+
+import junit.framework.TestCase;
+import org.apache.maven.project.MavenProject;
+import org.appformer.maven.support.AFReleaseId;
+import org.appformer.maven.support.AFReleaseIdImpl;
+import org.junit.Assert;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class ArtifactResolverTest extends TestCase {
+
+    public void testInJarResolverSnapshotLocal() throws MalformedURLException {
+
+        ClassLoader classLoader = getClass().getClassLoader();
+        AFReleaseId releaseId = new AFReleaseIdImpl("org.jbpm", "kjar", "1.0.0-SNAPSHOT", "jar");
+
+        InJarArtifactResolver inJarArtifactResolver = new MyInJarArtifactResolver(classLoader, releaseId, "SNAPSHOT");
+
+        Assert.assertTrue(inJarArtifactResolver.isLoaded());
+    }
+
+    public void testInJarResolverSnapshotExternal() throws MalformedURLException {
+
+        ClassLoader classLoader = getClass().getClassLoader();
+        AFReleaseId releaseId = new AFReleaseIdImpl("org.jbpm", "kjar", "1.0.0-SNAPSHOT", "jar");
+
+        InJarArtifactResolver inJarArtifactResolver = new MyInJarArtifactResolver(classLoader, releaseId, "20240717-143015-1");
+
+        Assert.assertTrue(inJarArtifactResolver.isLoaded());
+    }
+
+    private static class MyInJarArtifactResolver extends InJarArtifactResolver {
+        String SNAPSHOT;
+
+        public MyInJarArtifactResolver(ClassLoader classLoader, AFReleaseId releaseId, String snapshot) {
+            super(classLoader, releaseId);
+            SNAPSHOT = snapshot;
+        }
+
+        @Override
+        protected List<URL> buildResources(Predicate<String> predicate) {
+            ArrayList<URL> urlArrayList = new ArrayList<URL>();
+
+            if (predicate.test("BOOT-INF/classes/KIE-INF/lib/kjar.pom")) {
+                try {
+                    urlArrayList.add(new URL("file", "fatjar", "BOOT-INF/classes/KIE-INF/lib/kjar-1.0.0-" + SNAPSHOT + ".pom"));
```

Comment:
@albfan `buildResources` is called by super class constructor, so at this point `SNAPSHOT` is `null`. So in the both tests `testInJarResolverSnapshotLocal` and `testInJarResolverSnapshotExternal`, `effectivePoms` has `kjar-1.0.0-null.pom`. Tests passed, but it's not what you want to test?

Reply:
Initially I create two classes, and refactoring probably did a mistake. I think is better to refactor to mock class easily.

Reply:
I try to resolve again, adding two files on test resources. It almost do it, but still getUrl  fails to load the poms

Reply:
I did a trick on getURL, so this is not a great test, but we can discard test too